boot/zephyr/main.c

```diff
@@ -371,30 +372,62 @@ void zephyr_boot_log_stop(void)
         */
 
 #if defined(CONFIG_MCUBOOT_SERIAL) || defined(CONFIG_BOOT_USB_DFU_GPIO)
-static bool detect_pin(const char* port, int pin, uint32_t expected, int delay)
+
+#ifdef CONFIG_MCUBOOT_SERIAL
+#define BUTTON_0_DETECT_DELAY CONFIG_BOOT_SERIAL_DETECT_DELAY
+#else
+#define BUTTON_0_DETECT_DELAY CONFIG_BOOT_USB_DFU_DETECT_DELAY
+#endif
+
+
+#define BUTTON_0_NODE DT_ALIAS(bootloader_button0)
```

Comment:
> This is kind of a generic name. I am surprised the alias doesn't have "mcuboot" in the name.

This relates to a discussion we're having about configuring software specific features in devicetree. (See zephyr-devel; https://lists.zephyrproject.org/g/devel/message/8456). 

I suppose mcuboot could do the same thing, maybe with an `/mcuboot` node with compatible `mcuboot-configuration`. Then you could start putting whatever you want in there, not just GPIOs. Another option would be perhaps to use the `/options` node that @sjg20 mentioned in this response, since that's used by u-boot, with a quite similar use case: https://lists.zephyrproject.org/g/devel/message/8460

Reply:
Appling your request will mean that It will be not a DT alias anymore.